transport/http/outbound.go

```diff
@@ -339,7 +349,27 @@ func (o *Outbound) getPeerForRequest(ctx context.Context, treq *transport.Reques
 
 func (o *Outbound) createRequest(treq *transport.Request) (*http.Request, error) {
 	newURL := *o.urlTemplate
-	return http.NewRequest("POST", newURL.String(), treq.Body)
+	hreq, err := http.NewRequest("POST", newURL.String(), treq.Body)
+	if err != nil {
+		return nil, err
+	}
+	// grpc in, http out case
+	if treq.Transport == "grpc" {
+		// from https://tools.ietf.org/html/rfc7540#section-8.1.2.3
+		// For ":authority", ...
+		// An intermediary that converts an HTTP/2 request to HTTP/1.1 MUST
+		// create a Host header field if one is not present in a request by
+		// copying the value of the ":authority" pseudo-header field.
+		if v, ok := treq.Headers.Get(http2AuthorityPseudoHeader); ok {
+			hreq.Host = v
+		}
+		// strip all http2 pseudo-header fields
+		for _, k := range http2PseudoHeaders {
```

Comment:
hmm, is stripping all the psudo headers the right way? I don't there, is there any other translations needed. I haven't looked in details, but for example, do we need to translate `:path` to http/1 URL? similarly for others. 

I am interested in a general translation protocol between HTTP/2 (gRPC) to HTTP/1 here, if we allow it.

Reply:
I thought about the same thing. From RFC, it only mentioned the behavior for `:authority` specifically and other pseudo-header field's behavior were left undefined.